src/rgw/rgw_data_sync.cc

```diff
@@ -990,6 +990,7 @@ std::ostream& operator<<(std::ostream& out, const bucket_shard_str& rhs) {
 class RGWRunBucketSyncCoroutine : public RGWCoroutine {
   RGWDataSyncCtx *sc;
   RGWDataSyncEnv *sync_env;
+  const RGWContinuousLeaseCR* lease_cr;
```

Comment:
Are we sure that the data log cr doesn't exit and deletes lease_cr before all the lease_cr users finish?

Reply:
i guess that assumes that all intermediate coroutines are well-behaved, and either `call()` their children and block, or drain the children they `spawn()` before finishing. probably better to pass this down as `boost::intrusive_ptr<const RGWContinuousLeaseCR>` to be safe